localstack/services/events/events_starter.py

```diff
@@ -65,14 +65,15 @@ def handle_numeric_conditions(conditions: List[Any], value: float):
     return True
 
 
-# TODO: refactor/simplify
+# TODO: refactor/simplify!
 def filter_event_based_on_event_format(self, rule_name: str, event: Dict[str, Any]):
     def filter_event(event_pattern_filter: Dict[str, Any], event: Dict[str, Any]):
         for key, value in event_pattern_filter.items():
-            event_value = event.get(key.lower())
+            event_value = event.get(key.lower(), event.get(key))
```

Comment:
is the intent here to match keys in a case-agnostic way?

Reply:
That's right. Added a comment to the code, to make this a bit more clear. 👍